### DISCUSSION TITLE
FacadeAPI appears deprecated

### DISCUSSION THREAD

**Discussion by heatherita** [flowlogix/flowlogix#308] [Q&A] [answered]
As I work through the upgrade from flowlogix 2.3.` -> 4.0.7 I see that `com.flowlogix.jeedao.FacadeAPI` appears to be deprecated. Is this true? We rely heavily on this Interface. 

```

**Comment by lprimak** [comment, marked as answer]
`FacadeAPI` has been replaced by `DaoHelper` class.
This is to decouple EJBs from base classes and to use delegation vs. inheritance (better design)
Functionality is mostly the same

See https://github.com/flowlogix/flowlogix/blob/main/jakarta-ee/flowlogix-jee/src/test/java/com/flowlogix/jeedao/FacadeTest.java for an example

**Reply by heatherita** [reply]
ok so no more interface? 

**OLD:**
```
public interface MyTestFacadeLocal extends FacadeAPI<MyTest, Long>

public class MyTestFacade extends AbstractFacade<MyTest, Long> implements MyTestFacadeLocal
```
**NEW:**

My "Facade" class just uses a DaoHelper member now, as MyControl does below?

```
public class MyTestFacade {
    private final EntityManager em = mock(EntityManager.class, RETURNS_DEEP_STUBS);
    class MyControl {
        @Delegate
        final DaoHelper<Integer, Long> facade = DaoHelper.<Integer, Long>builder()
                .entityClass(Integer.class)
                .entityManagerSupplier(() -> em)
                .build();
```
What should I do about xa and non-xa EntityManager? What replaces this?

```
@Override
    protected EntityManager getEntityManager()
    {
        return isXA()? emXA : _em;
    }
```
I appreciate your responses thus far, but I need some guidance on how to port my old DAO architecture, comprised of "Local" interfaces that were implemented by classes that extend AbstractFacade,  to this new architecture or I may have to move away from Flowlogix to something more thoroughly documented. I have to meet a deadline.

**Reply by lprimak** [reply]
I would create my own Facade API (copy it from the old version) and use `@Delegate DaoHelper...` in your own AbstractFacade
that you can write. Those two are no longer in scope of the library. I have removed all BAW-specific stuff from it.

I'm sorry but I cannot give out free support. You have more than enough information to make this work.
I will answer any specific questions related directly to the library and it's functionality but I can't help with your project itself, not for free.

When I started, the low-code platforms for the web were not up to par, but, to be honest, I would look at low-code
tool for what you need and implement that. I think those tools are good enough now to get what you need done.
